src/AlgebraicGeometry/Schemes/ProjectiveSchemes/Objects/Attributes.jl

```diff
@@ -583,3 +583,44 @@ function relative_cotangent_module(X::AbsProjectiveScheme{<:Ring, <:MPolyQuoRing
   return cokernel(psi)
 end
 
+@doc raw"""
+    genus(X::AbsProjectiveScheme{<:Field}; algorithm::Symbol=:normalization)
+
+Given a projective curve `X` return the genus of `X`, i.e. the 
+integer `p_g = p_a - delta` where `p_a` is the arithmetic genus 
+and `delta` the delta-invariant of the curve.
+
+The `algorithm` keyword can be specified to 
+  - `:normalization` to go via the computation of a normalization
+  - `:primary_decomposition` to proceed with a primary decomposition
+Normalization is usually faster, but not always.
+"""
+function genus(X::AbsProjectiveScheme{<:Field}; algorithm::Symbol=:normalization)
+  @req dim(X) == 1 "scheme must be one-dimensional"
+  get_attribute!(X, :genus) do
```

Comment:
@ooinaruhugh : This is an example for how to write attribute functions with kwargs.

Reply:
`genus` and arithmetic genus are implemented already for `AbsCuve`.
It seems like it is hard discover?

Reply:
If you continue with this, then please change the interface there accordingly.

Reply:
What is genus if the curve is not reduced? I mean what is the delta invariant if the curve has multiplicities, say? (or fat points)